packages/icon/src/Icon.spec.tsx

```diff
@@ -1,20 +1,71 @@
 import React from 'react';
+import path from 'path';
+import fs from 'fs';
+import { toJson } from 'xml2json';
 import { render, cleanup } from '@testing-library/react';
 import { typeIs } from '@leafygreen-ui/lib';
 import { createIconComponent } from '.';
 
 afterAll(cleanup);
 
+describe('packages/Icon/glyphs/', () => {
+  const glyphPaths = fs
+    .readdirSync(path.resolve(__dirname, './glyphs'))
+    /**
+     * './glyphs/' contains an index.ts file, so we filter out
+     * anything that's not an svg.
+     */
+    .filter(path => /.*\.svg$/.test(path));
+
+  glyphPaths.forEach(glyphPath => {
+    describe(`${glyphPath}`, () => {
+      const { svg } = require(path.resolve(__dirname, `./glyphs/${glyphPath}`));
+
+      if (!svg) {
+        return;
+      }
+
+      type SVGNodeObject = { readonly [K in string]: string | SVGNodeObject };
+
+      const rootGlyphObject: SVGNodeObject = toJson(svg, { object: true });
+
+      let isValid = true;
+
+      function processGlyphObject(obj: SVGNodeObject) {
+        Object.keys(obj).forEach(key => {
+          const currentValue = obj[key];
+
+          if (typeof currentValue === 'object') {
+            processGlyphObject(currentValue);
+          } else {
+            if (key === 'fill') {
+              const validFills = ['#000', '#000000', 'none'];
+
+              if (!validFills.includes(currentValue)) {
```

Comment:
Could this not be the test itself? Such that if it fails the test shows what value it currently is? Maybe it doesn't matter?

Reply:
I did something that _almost_ does what you're referring to. Unfortunately you won't see the current value, as I couldn't find a way to do something like `expect(currentValue).toBeWithin(validFills)`. If you have a better idea, let me know, but the new implementation is an improvement.